src/Rotalia/APIBundle/Controller/PurchasesController.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Rotalia\APIBundle\Controller;
+
+use Exception;
+use Rotalia\InventoryBundle\Component\HttpFoundation\JSendResponse;
+use Rotalia\InventoryBundle\Model\Transaction;
+use Rotalia\InventoryBundle\Model\TransactionQuery;
+use Rotalia\InventoryBundle\Model\TransferQuery;
+use Rotalia\UserBundle\Model\User;
+use Symfony\Component\HttpFoundation\Request;
+use DateTime;
+use Nelmio\ApiDocBundle\Annotation\ApiDoc;
+
+
+class PurchasesController extends DefaultController
+{
+
+    /**
+     * Fetch list of purchases
+     *
+     * @param Request $request
+     * @ApiDoc(
+     *     resource = true,
+     *     statusCodes = {
+     *          200 = "Returned when successful",
+     *          400 = "Returned when dateFrom or dateUntil are not in valid date format",
+     *          403 = "Returned when user has insufficient privileges",
+     *     },
+     *     description="Fetch purchases list",
+     *     section="Purchase",
+     *     filters={
+     *          {"name"="conventId","type"="int","description"="Fetch purchases for another convent than member home convent"},
+     *          {"name"="memberId","dataType"="integer","description"="Member ID of user in interest"},
+     *          {"name"="dateFrom","type"="string","description":"datetime string"},
+     *          {"name"="dateUntil","type"="string","description":"datetime string"},
+     *          {"name"="limit","type"="int","description":"Limit number of purchases returned, default 5"},
+     *          {"name"="offset","type"="int","description":"Offset purchases returned for pagination, default 0"},
+     *     }
+     * )
+     *
+     * @param Request $request
+     * @return JSendResponse
+     */
+    public function listAction(Request $request)
+    {
+        $conventId = $request->get('conventId', null);
+        $memberId = $request->get('memberId', null);
+        $dateFrom = $request->get('dateFrom', null);
+        $dateUntil = $request->get('dateUntil', null);
+        $limit = $request->get('limit', 5);
+        $offset = $request->get('offset', 0);
+
+        $purchaseQuery = TransactionQuery::create()
```

Comment:
Hmm, miks selle controlleri nimeks sai PurchasesController muidu? Kui TransactionsQuery kaudu küsid objekte, siis oleks loogiline, et tegemist on TransactionsController'ga ju.

Reply:
Sinu loodud _transactionid_ on võimekamad. Ma lõin selle lahku _purchases_ ja _transfers_. Selle eristamiseks panin teise nime nii api endpointile kui ka muutujatele. Samas elan osaliselt ikkagi sinu _transactionite_ otsas edasi.